src/backend/access/zedstore/zedstore_tuplebuffer.c

```diff
@@ -138,10 +137,27 @@ get_tuplebuffer(Relation rel)
 }
 
 /*
- * Allocate a TID for insert.
+ * Allocate TIDs for insert.
+ *
+ * First check if the reserved tids can cater to the number of tids requested for
+ * allocation (ntids). If yes, consume the tids from these reserved tids. Else,
+ * we have to request more tids by inserting into the tid tree.
+ *
+ * We reserve tids inside the tupbuffer for the same (xid, cid) combo. The way we
+ * reserve tids is slightly different for single-insert vs multi-insert.
+ *
+ * For single inserts, in the same (xid, cid) once we encounter number of inserts =
+ * SINGLE_INSERT_TID_RESERVATION_THRESHOLD, we request and reserve
+ * SINGLE_INSERT_TID_RESERVATION_SIZE number of tids.
+ *
+ * For multi-inserts, we request and reserve (ntids * MULTI_INSERT_TID_RESERVATION_FACTOR)
+ * number of tids.
+ *
+ * TODO: expose these as GUCs as they are very workload sensitive (especially
```

Comment:
Should this TODO be moved to where `MULTI_INSERT_TID_RESERVATION_FACTOR` is defined

Reply:
Makes sense!